### PR TITLE
Add typescript n convert all files

### LINKED ISSUE #52: Add Typescript
Add typescript to project.

### PR DESCRIPTION
**Pull request by ayrbox**
1. Add typescript and @types packages.
1. Convert all js files to typescript.
1. Add type to all files (hooks, component and App.tsx)

Resolve #52

### CODE REVIEW

**Comment by ayrbox** [comment]
@JesseRWeigel 
It was my fault on earlier PR. My fork was not sync with main project. 

Let me know if any change is required. Thanks :+1:

File: package.json [hunk 1]
```diff
@@ -1,34 +1,52 @@
...
-    "not ie <= 11",
-    "not op_mini all"
-  ],
-  "devDependencies": {
-    "cypress": "^10.8.0",
-    "eslint-plugin-cypress": "^2.2.1"
-  }
-}
+{
+  "name": "test-app",
```
Comment (JesseRWeigel) [inline package.json:2]: Let's change this back to battlemath

**Review by JesseRWeigel — CHANGES_REQUESTED** [review]

**Review by JesseRWeigel — APPROVED** [review]

**Comment by JesseRWeigel** [comment]
Looks like the conflicts are back. I merged a branch with some css changes. That may have caused it.

**Comment by JesseRWeigel** [comment]
I made all the fixes for the merge conflicts but messed something up and can't update this branch. It made a new branch instead: [https://github.com/JesseRWeigel/battlemath/commit/50b74f43de96d9284ad9ad2de9dc09df4f7a2a34](https://github.com/JesseRWeigel/battlemath/commit/50b74f43de96d9284ad9ad2de9dc09df4f7a2a34)

**Comment by ayrbox** [comment]
@JesseRWeigel sorry, I was not able fix the conflict early. I've fixed now. Please merge it if its good enough :)

**Comment by JesseRWeigel** [comment]
Thanks @ayrbox !

**Pull request merged**